chart/templates/statefulset.yaml

```diff
@@ -0,0 +1,121 @@
+{{- if .Values.statefulset.enabled -}}
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "kube-httpcache.fullname" . }}
+  labels:
+    {{- include "kube-httpcache.labels" . | nindent 4 }}
+spec:
+  serviceName: {{ include "kube-httpcache.fullname" . }}
+{{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+{{- end }}
+  selector:
+    matchLabels:
+      {{- include "kube-httpcache.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+    {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+    {{- end }}
+      labels:
+        {{- include "kube-httpcache.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "kube-httpcache.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      {{- if .Values.extraInitContainers }}
+      initContainers:
+        {{- with .Values.extraInitContainers }}
+        {{- tpl . $ | nindent 8 }}
+        {{- end }}
+      {{- end }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+          - -admin-addr=0.0.0.0
+          - -admin-port=6083
+          - -signaller-enable
+          - -signaller-port=8090
+          - -frontend-watch
+          - -frontend-namespace=$(NAMESPACE)
+          - -frontend-service={{ .Values.cache.frontendservice }}
+          - -backend-watch
+          - -backend-namespace=$(NAMESPACE)
+          - -backend-service={{ .Values.cache.backendservice }}
+          - -varnish-secret-file=/etc/varnish/k8s-secret/secret
+          - -varnish-vcl-template=/etc/varnish/tmpl/default.vcl.tmpl
+          - -varnish-storage={{ .Values.cache.varnishstorage }},{{ .Values.cache.storagesize }}
+          {{- if .Values.cacheExtraArgs }}
+          {{- with .Values.cacheExtraArgs }}
+          {{- tpl . $ | trim | nindent 10 }}
+          {{- end }}
+          {{- end }}
+          env:
+          - name: NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+          volumeMounts:
+          - name: template
+            mountPath: /etc/varnish/tmpl
+          - name: secret
+            mountPath: /etc/varnish/k8s-secret
+          {{- if .Values.extraMounts }}
+            {{- with .Values.extraMounts }}
+            {{- tpl . $ | trim | nindent 10 }}
+            {{- end }}
+          {{- end }}
+          {{- if .Values.livenessprobe }}
+          livenessProbe:
+            {{- with .Values.livenessprobe }}
+            {{- tpl . $ | trim | nindent 12 }}
```

Comment:
Is there a particular reason to favor `tpl` over `toYaml` here?

I currently can't think of an use-case where it would be useful to template the `livenessProbe` (or `extraVolumes` and so on).

This seems to come at the cost of additional complexity, as the matching entries in the `values.yaml` are now strings and single fields can't easily be extended/overwritten by merging multiple `values.yaml` files.

Reply:
:+1: on this. Until @hensur mentioned it, I didn't even _notice_ that the values in the `values.yaml` weren't supposed to be structured YAML, but _strings_ with YAML content instead. I think this might be highly surprising for the unsuspecting user.